core/src/main/java/com/graphhopper/routing/lm/LMPreparationHandler.java

```diff
@@ -218,37 +217,38 @@ public int size() {
         return preparations;
     }
 
-    @Override
-    public RoutingAlgorithmFactory getDecoratedAlgorithmFactory(RoutingAlgorithmFactory defaultAlgoFactory, HintsMap map) {
-        // for now do not allow mixing CH&LM #1082
-        boolean disableCH = map.getBool(Parameters.CH.DISABLE, false);
-        boolean disableLM = map.getBool(Parameters.Landmark.DISABLE, false);
-        if (!isEnabled() || disablingAllowed && disableLM || !disableCH)
-            return defaultAlgoFactory;
-
+    /**
+     * @return a {@link RoutingAlgorithmFactory} for LM or throw an error if no preparation is available for the given
+     * hints
+     */
+    public RoutingAlgorithmFactory getAlgorithmFactory(HintsMap map) {
         if (preparations.isEmpty())
-            throw new IllegalStateException("No preparations added to this decorator");
+            throw new IllegalStateException("No LM preparations added yet");
 
         // if no weighting or vehicle is specified for this request and there is only one preparation, use it
         if ((map.getWeighting().isEmpty() || map.getVehicle().isEmpty()) && preparations.size() == 1) {
-            return new LMRAFactory(preparations.get(0), defaultAlgoFactory);
+            return new LMRAFactory(preparations.get(0), new RoutingAlgorithmFactorySimple());
         }
 
+        List<Weighting> lmWeightings = new ArrayList<>(preparations.size());
         for (final PrepareLandmarks p : preparations) {
+            lmWeightings.add(p.getWeighting());
             if (p.getWeighting().matches(map))
-                return new LMRAFactory(p, defaultAlgoFactory);
+                return new LMRAFactory(p, new RoutingAlgorithmFactorySimple());
         }
 
+        // todonow: what does that mean?
         // if the initial encoder&weighting has certain properties we could cross query it but for now avoid this
```

Comment:
This comment can probably be removed, but I was not sure what 'cross query it' means here?

Reply:
It means if we have a LM preparation for e.g. "fastest+car" and it should be okay to do a request also with a changed or even different weighting (for optimality the edge weights must not decrease and so not all changes are acceptable).